### PR TITLE
LPS-35843

### PR DESCRIPTION
**Pull request by jbalsas**

### CODE REVIEW

File: portal-web/docroot/html/js/liferay/util.js [hunk 1]
```diff
@@ -1622,14 +1622,25 @@
...
+				visibleChangeHandle = dialog.after('visibleChange', detachSelectionEventFn);
+
 				dialog.show();
 			}
 			else {
 				Util.openWindow(
 					config,
 					function(dialogWindow) {
-						dialogWindow.after('close', selectionEvent.detach, selectionEvent);
+						visibleChangeHandle = dialogWindow.after('visibleChange', detachSelectionEventFn);
```
Comment (jbalsas) [inline portal-web/docroot/html/js/liferay/util.js:1643]: @eduardolundgren @ipeychev Is `visibleChange` the only way to detect this now? Shouldn't we be firing a `hide` event? Also, maybe even a `close` event for backwards compatibility?

**Comment by eduardolundgren** [comment]
Hey Chema, the old dialog have 'hide', 'close', 'visibleChange' and 'destroy' custom events that were being used to add logic when the dialog hides os gets destroyed. To keep consistence and simplify API confusion we no longer have those events, we should rely on visibleChange only, it's more verbose although it's cleaner as a long term.

If you really want to push other custom events like before, please, feel free to ping me and we can discuss, I am open to that.

Thanks!

**Comment by eduardolundgren** [comment]
Hey Chema, I have reviewed the code and sent to Brian here https://github.com/brianchandotcom/liferay-portal/pull/11516, could you re-test when it gets in the portal?

Thank you!

**Comment by jbalsas** [comment]
Hey Eduardo, this landed last night on master and I've just verified it works properly. Thanks!

About the custom events, it's okay, I just wanted to be sure there was no cleaner way to do it. 

I would usually argue that a more _semantic_ API (with custom `hide` and `show` events) is bound to be more developer-friendly in the long term, though. I'd assume every time you listen for `visibleChange` you'd want to know wether it showed or hid, so having a named event for that would be easier. Just a thought ;)

**Comment by eduardolundgren** [comment]
Right. Let's keep that in mind for discussion. Thanks.

**Pull request closed without merging**